instrumentation/log4j/log4j-2.13.2/library/src/main/java/io/opentelemetry/instrumentation/log4j/v2_13_2/LogEventMapper.java

```diff
@@ -0,0 +1,141 @@
+import io.opentelemetry.api.trace.SpanId;
+import io.opentelemetry.api.trace.TraceFlags;
+import io.opentelemetry.api.trace.TraceId;
+import io.opentelemetry.api.trace.TraceState;
+import io.opentelemetry.context.Context;
+import io.opentelemetry.sdk.logs.LogBuilder;
+import io.opentelemetry.sdk.logs.data.Severity;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+import org.apache.logging.log4j.Level;
+import org.apache.logging.log4j.Marker;
+import org.apache.logging.log4j.core.LogEvent;
+import org.apache.logging.log4j.message.Message;
+import org.apache.logging.log4j.util.ReadOnlyStringMap;
+
+final class LogEventMapper {
+
+  // Visible for testing
+  static final AttributeKey<String> ATTR_LOGGER_NAME = AttributeKey.stringKey("logger.name");
+  static final AttributeKey<String> ATTR_THREAD_NAME = AttributeKey.stringKey("thread.name");
+  static final AttributeKey<Long> ATTR_THREAD_ID = AttributeKey.longKey("thread.id");
+  static final AttributeKey<Long> ATTR_THREAD_PRIORITY = AttributeKey.longKey("thread.priority");
+  static final AttributeKey<String> ATTR_THROWABLE_NAME = AttributeKey.stringKey("throwable.name");
+  static final AttributeKey<String> ATTR_THROWABLE_MESSAGE =
+      AttributeKey.stringKey("throwable.message");
+  static final AttributeKey<List<String>> ATTR_NDC = AttributeKey.stringArrayKey("ndc");
+  static final AttributeKey<String> ATTR_FQCN = AttributeKey.stringKey("fqcn");
+  static final AttributeKey<String> ATTR_MARKER = AttributeKey.stringKey("marker");
+
+  private static final Map<Level, Severity> LEVEL_SEVERITY_MAP;
+
+  static {
+    Map<Level, Severity> levelSeverityMap = new HashMap<>();
+    levelSeverityMap.put(Level.ALL, Severity.TRACE);
+    levelSeverityMap.put(Level.TRACE, Severity.TRACE2);
+    levelSeverityMap.put(Level.DEBUG, Severity.DEBUG);
+    levelSeverityMap.put(Level.INFO, Severity.INFO);
+    levelSeverityMap.put(Level.WARN, Severity.WARN);
+    levelSeverityMap.put(Level.ERROR, Severity.ERROR);
+    levelSeverityMap.put(Level.FATAL, Severity.FATAL);
+    LEVEL_SEVERITY_MAP = Collections.unmodifiableMap(levelSeverityMap);
+  }
+
+  static void mapLogEvent(LogBuilder builder, LogEvent logEvent) {
+    AttributesBuilder attributes = Attributes.builder();
+
+    // message
+    Message message = logEvent.getMessage();
+    if (message != null) {
+      builder.setBody(message.getFormattedMessage());
+    }
+
+    // time
+    builder.setEpoch(logEvent.getNanoTime(), TimeUnit.NANOSECONDS);
+
+    // level
+    Level level = logEvent.getLevel();
+    if (level != null) {
+      builder.setSeverity(
+          LEVEL_SEVERITY_MAP.getOrDefault(level, Severity.UNDEFINED_SEVERITY_NUMBER));
+      builder.setSeverityText(logEvent.getLevel().name());
+    }
+
+    // logger
+    attributes.put(ATTR_LOGGER_NAME, logEvent.getLoggerName());
+
+    // fully qualified class name
```

Comment:
Hmm - I wonder what this is, I thought the only "class name" type of thing is the logger name, if you derive it from a class. Otherwise each log message would be quite expensive

Reply:
This [test](https://github.com/open-telemetry/opentelemetry-java-instrumentation/pull/4375/files#diff-c3aa4475c1045c52b212b54274b73f836e2cdca46381c0dd2e5601e87243bea9R135) demonstrates the value. In that  case it was `org.apache.logging.log4j.spi.AbstractLogger`. Not sure how useful it is, but as mentioned [here](https://github.com/open-telemetry/opentelemetry-java-instrumentation/pull/4375#discussion_r749527440), my stance was to map everything.